File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.js
/**
 * @param {string} s
 * @return {boolean}
 */
var validPalindrome = function(s) {
    const backtrack = (s, p, q, deleted) => {
        // console.log(p, " : ", q, " : deleted: ", deleted)
        if(q <= p) return true
        if(s[p] === s[q]) {
            // console.log("ch: ", s[p], " : ", p+1, " : ", q-1)
            return backtrack(s, p+1, q-1, deleted)
        } else {
            if(deleted) return false
            return backtrack(s, p+1, q, true) || backtrack(s, p, q-1, true)
        }
    }
    return backtrack(s, 0, s.length-1, false)
};

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.java
class Solution {
    boolean backtrack(String s, int l, int r, int k) {
        if(r <= l) return true;

        if(s.charAt(l) == s.charAt(r)) {
            return backtrack(s, l+1, r-1, k);
        } else {
            if(k > 0) {
                return backtrack(s, l+1, r, k-1) || backtrack(s, l, r-1, k-1);
            } else {
                return false;
            }
        }
    }
    public boolean validPalindrome(String s) {
        return backtrack(s, 0, s.length()-1, 1);
    }
}